test/testdata/resolver/recover_from_bad_superclass.rb

```diff
@@ -0,0 +1,37 @@
+# typed: true
+
+def make_a_class
+  Class.new
+end
+
+class A < 1 # error: Superclasses must only contain constant literals
+end
+T.let(A, Class)
+A.new.class
```

Comment:
What is this line testing?

Reply:
I'm curious to see what `T.reveal_type` says for this.

Reply:
It was mostly for the `new` call since before this `new` didn't work in these cases. However I think this is covered by the `let` assertion. I will remove them